### PR TITLE
Support bulk exports in revamped UX

### PR DESCRIPTION
**Pull request by nickpell**
Adds link from "Bulk Download of Form Export" button...
screen shot 2015-07-07 at 5 27 00 pm
... to download page with filters ...
screen shot 2015-07-07 at 5 27 08 pm
... where you can download the export ...
screen shot 2015-07-07 at 5 27 13 pm
(can ignore error message in screenshot)

Much of the existing code will become dead code once the feature flag is lifted.  I'll delete that code then.

Sorry for the messy commit history.

@NoahCarnahan cc: @orangejenny

### CODE REVIEW

**Comment by dimagimon** [comment]
There are 17 errors:

* corehq/apps/reports/static/reports/ko/export.manager.js, line 239 - ['xmlns'] is better written in dot notation.
* corehq/apps/reports/static/reports/ko/export.manager.js, line 240 - ['modulename'] is better written in dot notation.
* corehq/apps/reports/static/reports/ko/export.manager.js, line 241 - ['exporttype'] is better written in dot notation.
* corehq/apps/reports/static/reports/ko/export.manager.js, line 242 - ['formname'] is better written in dot notation.
* corehq/apps/reports/static/reports/ko/export.manager.js, line 266 - The array literal notation [] is preferable.
* corehq/apps/reports/static/reports/ko/export.manager.js, line 274 - '_id' is already defined.
* corehq/apps/reports/static/reports/ko/export.manager.js, line 275 - 'xmlns' is already defined.
* corehq/apps/reports/static/reports/ko/export.manager.js, line 276 - 'module' is already defined.
* corehq/apps/reports/static/reports/ko/export.manager.js, line 277 - 'export_type' is already defined.
* corehq/apps/reports/static/reports/ko/export.manager.js, line 278 - 'form' is already defined.
* corehq/apps/reports/static/reports/ko/export.manager.js, line 280 - 'sheetName' is already defined.
* corehq/apps/reports/static/reports/ko/export.manager.js, line 289 - 'export_tag' is already defined.
* corehq/apps/reports/static/reports/ko/export.manager.js, line 308 - The array literal notation [] is preferable.
* corehq/apps/reports/static/reports/ko/export.manager.js, line 436 - Missing semicolon.
* corehq/apps/reports/static/reports/ko/export.manager.js, line 441 - Missing semicolon.
* corehq/apps/reports/static/reports/ko/export.manager.js, line 451 - Missing semicolon.
* corehq/apps/reports/static/reports/ko/export.manager.js, line 456 - Missing semicolon.

**Comment by dimagimon** [comment]
There are 8 errors:

* corehq/apps/reports/static/reports/ko/export.manager.js, line 274 - '_id' is already defined.
* corehq/apps/reports/static/reports/ko/export.manager.js, line 275 - 'xmlns' is already defined.
* corehq/apps/reports/static/reports/ko/export.manager.js, line 276 - 'module' is already defined.
* corehq/apps/reports/static/reports/ko/export.manager.js, line 277 - 'export_type' is already defined.
* corehq/apps/reports/static/reports/ko/export.manager.js, line 278 - 'form' is already defined.
* corehq/apps/reports/static/reports/ko/export.manager.js, line 280 - 'sheetName' is already defined.
* corehq/apps/reports/static/reports/ko/export.manager.js, line 289 - 'export_tag' is already defined.
* corehq/apps/reports/static/reports/ko/export.manager.js, line 308 - The array literal notation [] is preferable.

**Comment by nickpell** [comment]
remaining code with dimagimon errors is on death row, so whatever.

File: corehq/apps/reports/standard/export.py [hunk 1]
```diff
@@ -431,11 +439,38 @@ class FormExportReport(FormExportReportBase):
...
         context.update({
-            'export': SavedExportSchema.get(export_id),
-            'additional_params': 'export_id=%(export_id)s' % {
-                'export_id': export_id,
-            },
+            'export': self.exports[0],
+            'exports': self.exports,
+            "use_bulk": len(self.export_ids) > 1,
+            'additional_params': mark_safe(
+                '&'.join('export_id=%(export_id)s' % {
```
Comment (orangejenny) [inline corehq/apps/reports/standard/export.py]: Python can handle this, yes, but creating get strings with multiple values for one param is...weird.

File: corehq/apps/reports/static/reports/ko/export.manager.js [hunk 2]
```diff
@@ -205,48 +233,85 @@ var ExportManager = function (o) {
...
+                var xmlns = export_data.xmlns,
+                    module = export_data.modulename,
+                    export_type = export_data.exporttype,
+                    form = export_data.formname,
+                    _id = export_id;
+
+                var sheetName = getSheetName(module, form, xmlns);
+
+                var export_tag;
+                if (self.is_custom)
```
Comment (orangejenny) [inline corehq/apps/reports/static/reports/ko/export.manager.js]: braces would make this if in particular easier to read

File: corehq/apps/reports/templates/reports/reportdata/data_export.html [hunk 3]
```diff
@@ -2,6 +2,35 @@
...
+            bulkDownloadPageUrlRoot: '{{ download_page_url_root }}'
+        });
+        ko.applyBindings(exportManager, $('#export-ui').get(0));
+
+        var bulkExportTitle = "{% trans "What is Bulk Export?" %}";
+        var bulkExportContent = (
+            "{% trans "Bulk Export creates a single Excel file with each export as a separate sheet inside that file." %}"
+            + "<br /><br />{% trans "NOTE: Excel 2007+ is currently the only bulk export format supported." %}"
+        );
+        $('.bulk-export-info').popover({
```
Comment (orangejenny) [inline corehq/apps/reports/templates/reports/reportdata/data_export.html]: Super minor, but this seems like help text as much as info text & could use the same help pattern we use most other places (hq-help-template)

File: corehq/apps/reports/standard/export.py (hunk 1 above)
Comment (czue) [inline corehq/apps/reports/standard/export.py]: true, though looks like there's no defined standard for it: http://stackoverflow.com/questions/24059773/correct-way-to-pass-multiple-values-for-same-parameter-name-in-get-request

File: corehq/apps/reports/static/reports/ko/export.manager.js (hunk 2 above)
Comment (czue) [inline corehq/apps/reports/static/reports/ko/export.manager.js]: +1 i would say we should almost treat braces as a requirement.

**Comment by czue** [comment]
(didn't review closely, was just lurking - @orangejenny feel free to merge whenever you're happy)

**Comment by czue** [comment]
(i also kicked build since it didn't look related)

**Comment by NoahCarnahan** [comment]
Just wanted to check: you're still iterating on the "download page with filters," right?

**Comment by nickpell** [comment]
build is failing with the common error:
```OperationalError: database "test_commcarehq" is being accessed by other users```
so I'm not worried

**Comment by nickpell** [comment]
@NoahCarnahan That's correct

File: corehq/apps/reports/static/reports/ko/export.manager.js (hunk 2 above)
Comment (nickpell) [inline corehq/apps/reports/static/reports/ko/export.manager.js]: agreed, but this code also will become dead code when the flags are lifted

**Comment by nickpell** [comment]
bump

**Pull request merged**